linkml/generators/fhirgen.py

```diff
@@ -0,0 +1,396 @@
+from linkml_runtime.utils.schemaview import SchemaView
+# from linkml.utils.generator import Generator, shared_arguments
+# todo: Do I need _ensure_ranked? If so, move out of docgen?
+# from linkml.generators.docgen import _ensure_ranked
+from linkml.utils.generator import Generator, shared_arguments
+
+# todo: make some of these class properties?
+FHIRGEN_DIR = os.path.join(os.path.dirname(__file__), 'fhirgen')
+MODEL_MAPPING_PATH = os.path.join(FHIRGEN_DIR, 'mapping__enum_valueset.csv')
+SUPPORTED_FORMATS = ['json']  # todo: later: xml
+FHIR_VALUESET_FIELD_MUTATIONS: Dict[str, Callable] = {
+    'status': lambda x: 'unknown'  # todo: temp
+}
+
+
+# TODO: when rebasing, this will be a dataclass. outdir will become a standard dataclass obj for example. don't make
+#  ...any of these changes before the dataclass PR is merged. (https://github.com/linkml/linkml/pull/924)
+class FhirValueSetGenerator(Generator):
+    """Generate FHIR ValueSet JSON from a LinkML YAML Enum schema"""
+    generatorname = os.path.basename(__file__)
+    generatorversion = "0.0.1"
+    valid_formats = SUPPORTED_FORMATS
+
+    def __init__(
+        self,
+        schema: Union[str, TextIO, SchemaDefinition],
+        outdir: str = None,
+        format: str = None,
+    ) -> None:
+        """
+        Creates a generator object that can write output
+
+        :param schema: path to schema file or schema object
+        :param outdir: path to directory save file or file(s)
+        :param genmeta:
+        """
+        self.sourcefile = schema
+        self.schemaview = SchemaView(schema)
+        self.schema = self.schemaview.schema
+        self.outdir = outdir
+        self.format = format
+
+    def serialize(self, save: bool = True, outdir: str = None, save_format='json') -> List[Dict]:
+        """
+        Serialize Enums in schema to JSON
+
+        :param save: if True, saves to disk.
+        :param outdir: path to directory save file or file(s)
+        :param save_format: Format to save.
+        :return: List of enums as dictionaries.
+        """
+        if save_format not in SUPPORTED_FORMATS:
+            raise ValueError(f'Format {save_format} not supported. Use one of: {",".join(SUPPORTED_FORMATS)}')
+        _outdir = outdir if outdir else self.outdir
+        if save and not _outdir:
+            raise RuntimeError('Tried to save, but did not supply `outdir`.')
+        sv = self.schemaview
+        linkml_dict_enums: Dict[str, EnumDefinition] = sv.all_enums()
+        fhir_dict_enums = []
+        for en, e in linkml_dict_enums.items():
+            d: Dict = self.serialize_enum(en, e)
+            fhir_dict_enums.append(fhir_dict_enums)
+            if save and save_format == 'json':
+                out_str: str = json.dumps(d)
+                self._write(out_str=out_str, outdir=_outdir, name=en, format=save_format)
+        return fhir_dict_enums
+
+
+    @staticmethod
+    def model_mappings(path: str = MODEL_MAPPING_PATH) -> Dict[str, str]:
```

Comment:
can you provide more details of what this is, and include an example in the unit test?

Also, it may be the case that this is not necessary - in theory the schema should provide all necessary information and mappings, no external out of band data should be required

Reply:
Hey Chris. It seemed like this feature might have been a priority at the time, but since September it's been pretty far down the list of things needed for TIMS and TermHub. But eventually it will be useful for both.

This PR was heavily in flux. I was going to try out schema automator and some other stuff. It is in need of unit testing and a lot of other work.

If you want, we can close this one right now. I can always re open at any time.